extensions/data-workspace/src/dialogs/projectDashboard.ts

```diff
@@ -186,54 +204,62 @@ export class ProjectDashboard {
 				.component();
 			tableContainer.addItem(tableNameLabel, { CSSStyles: { 'padding-left': '25px', 'padding-bottom': '20px', ...constants.cssStyles.title } });
 
-			const columns: azdata.DeclarativeTableColumn[] = [];
-			info.columns.forEach((column: IDashboardColumnInfo) => {
-				let col = {
-					displayName: column.displayName,
-					valueType: column.type === 'icon' ? azdata.DeclarativeDataType.component : azdata.DeclarativeDataType.string,
-					isReadOnly: true,
-					width: column.width,
-					headerCssStyles: {
-						'border': 'none',
-						...constants.cssStyles.tableHeader
-					},
-					rowCssStyles: {
-						...constants.cssStyles.tableRow
-					},
-				};
-				columns.push(col);
-			});
-
-			const data: azdata.DeclarativeTableCellValue[][] = [];
-			info.data.forEach(values => {
-				const columnValue: azdata.DeclarativeTableCellValue[] = [];
-				values.forEach(val => {
-					if (typeof val === 'string') {
-						columnValue.push({ value: val });
-					} else {
-						const iconComponent = this.modelView!.modelBuilder.image().withProperties<azdata.ImageComponentProperties>({
-							iconPath: val.icon,
-							width: '15px',
-							height: '15px',
-							iconHeight: '15px',
-							iconWidth: '15px'
-						}).component();
-						const stringComponent = this.modelView!.modelBuilder.text().withProperties<azdata.TextComponentProperties>({
-							value: val.text,
-							CSSStyles: { 'margin-block-start': 'auto', 'block-size': 'auto', 'margin-block-end': '0px' }
-						}).component();
-
-						const columnData = this.modelView!.modelBuilder.flexContainer().withItems([iconComponent, stringComponent], { flex: '0 0 auto', CSSStyles: { 'margin-right': '10px' } }).withLayout({ flexFlow: 'row' }).component();
-						columnValue.push({ value: columnData });
-					}
+			if (info.data.length === 0) {
+				const noDataText = constants.noPreviousData(info.name.toLocaleLowerCase());
+				const noDataLabel = this.modelView!.modelBuilder.text()
+					.withProperties<azdata.TextComponentProperties>({ value: noDataText })//, CSSStyles: { 'margin-block-start': '30px', 'margin-block-end': '0px' } })
+					.component();
+				tableContainer.addItem(noDataLabel, { CSSStyles: { 'padding-left': '25px', 'padding-bottom': '20px' } });
+			} else {
+				const columns: azdata.DeclarativeTableColumn[] = [];
+				info.columns.forEach((column: IDashboardColumnInfo) => {
+					let col = {
+						displayName: column.displayName,
+						valueType: column.type === 'icon' ? azdata.DeclarativeDataType.component : azdata.DeclarativeDataType.string,
+						isReadOnly: true,
+						width: column.width,
+						headerCssStyles: {
+							'border': 'none',
```

Comment:
these changes are only showing because of the added spaced dues to if condition right? or there are real changes here?

Reply:
Due to if condition. No real changes here :)